src/main/java/org/broadinstitute/hellbender/tools/exome/AnnotateTargets.java

```diff
@@ -112,41 +198,145 @@ public Object onTraversalSuccess() {
+
+        /**
+         * Lazy annotators must implement this function
+         *
+         * @param targets a list of traversed targets
+         * @param rawAnnotations a list of collected "raw" annotations
+         * @return processed annotations
+         */
+        default List<String> processAnnotations(final List<Target> targets, final List<Object> rawAnnotations) {
+            throw new UnsupportedOperationException("Either this annotator is not lazy, or the annotation post-processing" +
+                    " method is not implemented yet");
+        }
     }
 
     private class GCContentAnnotator implements TargetAnnotator {
         @Override
-        public Double apply(final Target target, final ReadsContext readsContext, final ReferenceContext referenceContext, final FeatureContext featureContext) {
+        public Double apply(final Target target, final ReadsContext readsContext, final ReferenceContext referenceContext,
+                            final FeatureContext featureContext) {
             final Nucleotide.Counter counter = new Nucleotide.Counter();
             counter.addAll(referenceContext.getBases());
             final long gcCount = counter.get(Nucleotide.C) + counter.get(Nucleotide.G);
             final long atCount = counter.get(Nucleotide.A) + counter.get(Nucleotide.T);
             final long totalCount = gcCount + atCount;
             return totalCount == 0 ? Double.NaN : gcCount / (double) totalCount;
         }
+
+        @Override
+        public boolean isLazy() {
+            return false;
+        }
+    }
+
+    private class BaitCountAnnotator implements TargetAnnotator {
+        @Override
+        public Set<SimpleInterval> apply(final Target target, final ReadsContext readsContext, final ReferenceContext referenceContext,
+                                         final FeatureContext featureContext) {
+            return featureContext.getValues(baits).stream().map(Target::getInterval).collect(Collectors.toSet());
+        }
+
+        @Override
+        public boolean isLazy() {
+            return true;
+        }
+
+        @Override
+        @SuppressWarnings("unchecked")
+        public List<String> processAnnotations(final List<Target> targets, final List<Object> rawAnnotations) {
+            if (targets.isEmpty()) {
+                return new ArrayList<>();
+            }
+            final List<Set<SimpleInterval>> overlappingBaitSetPerTarget = rawAnnotations.stream()
+                    .map(rawAnnotation -> (Set<SimpleInterval>) rawAnnotation)
+                    .collect(Collectors.toList());
+            final double[] baitCounts = calculateBaitCounts(targets, overlappingBaitSetPerTarget);
+            return Arrays.stream(baitCounts).mapToObj(String::valueOf).collect(Collectors.toList());
+        }
+    }
+
+    @VisibleForTesting
+    static double[] calculateBaitCounts(List<Target> targets, List<Set<SimpleInterval>> overlappingBaitSetPerTarget) {
```

Comment:
This method needs a unit test, since it is used to calculate the expected output of the integration test.